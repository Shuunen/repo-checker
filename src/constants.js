import { join } from 'path'

export const home = process.env.HOME
export const dataFileName = 'repo-checker.config.js'
export const repoCheckerPath = join(__dirname, '..')
export const templatePath = join(repoCheckerPath, 'templates')
export const dataFileHomePath = join(home, dataFileName)

export const dataDefaults = {
  auto_merge: true,
  ban_sass: true,
  dev_deps_only: true,
  license: 'GPL-3.0',
  max_size_ko: 10,
  user_mail: 'romain.racamier@gmail.com',
  user_name: 'Romain Racamier-Lafon',
}
