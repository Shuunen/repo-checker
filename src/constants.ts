import { join } from 'path'

/* istanbul ignore next */
export const home = process.env.HOME ?? '' // when does HOME is not defined ?
export const dataFileName = 'repo-checker.config.js'
/* istanbul ignore next */
export const repoCheckerPath = process.env.pwd ?? process.cwd()
export const templatePath = join(repoCheckerPath, 'templates')
export const dataFileHomePath = join(home, dataFileName)

export class ProjectData {
  auto_merge = true
  ban_sass = true
  dev_deps_only = true
  is_module = false
  license = 'GPL-3.0'
  max_size_ko = 1
  npm_package = false
  package_name = ''
  quiet = false
  repo_id = 'a-great-repo'
  use_typescript = false
  use_vue = false
  user_id = 'Shuunen'
  user_id_lowercase = 'shuunen'
  user_mail = 'romain.racamier@gmail.com'
  user_name = 'Romain Racamier-Lafon'
  web_published = false
  web_url = 'https://my-website.com'

  constructor (data: Partial<ProjectData> = {}) {
    Object.assign(this, data)
  }
}

export const dataDefaults = new ProjectData()
