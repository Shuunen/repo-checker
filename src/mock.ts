/* eslint-disable jsdoc/require-jsdoc */
import path from 'node:path'
import { clone, sleep } from 'shuutils'
import type { Indicators } from './check.ts'
import type { FileBase } from './file.ts'
import { join } from './utils.ts'

const fileBaseKeysToDelete: (keyof FileBase)[] = ['fileExists', 'checkFileExists', 'folderPath', 'inspectFile', 'updateFile', 'initFile']
const indicatorsKeys: ReadonlyArray<keyof Indicators> = ['failed', 'passed', 'warnings'] as const

function cleanStringForSnap(input: string) {
  return (
    input
      // replace repo-check-1-40 by repo-check-x-yy
      .replace(/repo-check-\d-\d+/gu, 'repo-check-x-yy')
      // replace package-json-main-file-size-63ko by package-json-main-file-size-XYZko
      .replace(/package-json-main-file-size-\d+ko/gu, 'package-json-main-file-size-XYZko')
  )
}

// eslint-disable-next-line no-restricted-syntax
export function cleanUnknownValueForSnap<Type>(input: Type): Type {
  /* c8 ignore next 8 */
  /* eslint-disable @typescript-eslint/consistent-type-assertions, @typescript-eslint/no-unsafe-return, @typescript-eslint/no-unsafe-type-assertion */
  if (typeof input === 'string') return cleanStringForSnap(input) as Type
  if (Array.isArray(input)) return input.map(item => cleanUnknownValueForSnap(item)) as Type
  if (typeof input === 'object') return JSON.parse(cleanStringForSnap(JSON.stringify(input))) as Type
  // eslint-disable-next-line no-restricted-syntax
  throw new Error(`cleanInstanceValueForSnap: unknown type ${typeof input}`)
  /* eslint-enable @typescript-eslint/consistent-type-assertions, @typescript-eslint/no-unsafe-return, @typescript-eslint/no-unsafe-type-assertion */
}

export async function promiseValue<Type>(value: Type) {
  await sleep(1)
  return value
}

export async function promiseTrue() {
  await sleep(1)
  return true
}

export async function promiseFalse() {
  await sleep(1)
  return false
}

export async function promiseVoid() {
  await sleep(1)
}

// eslint-disable-next-line unicorn/prefer-module
export const sourceFolder = path.dirname(__filename)

export const vueProjectFolder = join(sourceFolder, 'mocks', 'vueProject')

export const tsProjectFolder = join(sourceFolder, 'mocks', 'tsProject')

export const mocksProjectsFolder = join(sourceFolder, 'mocks')

export function cleanIndicatorsForSnap(record?: Readonly<Record<string, unknown>>) {
  if (!record) return record
  const clean = clone(record)
  for (const key of indicatorsKeys) if (key in clean) clean[key] = cleanUnknownValueForSnap(clean[key])
  return clean
}

// eslint-disable-next-line @typescript-eslint/prefer-readonly-parameter-types
export function cleanInstanceForSnap(instance: Readonly<FileBase>, ...bonusKeys: readonly string[]) {
  // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
  const clean = clone<Partial<FileBase>>(instance) as Record<string, unknown>
  const keys = [...fileBaseKeysToDelete, ...bonusKeys] // eslint-disable-next-line @typescript-eslint/no-dynamic-delete
  for (const key of keys) if (key in clean) delete clean[key]
  return cleanIndicatorsForSnap(clean)
}
