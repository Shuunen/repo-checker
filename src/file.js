import path from 'path'
import { templatePath } from './constants'
import { log } from './logger'
import { createFile, fillTemplate, folderContainsFile, getFileSizeInKo, readFileInFolder } from './utils'

export class File {
  constructor (folderPath, data, doFix, doForce) {
    this.folderPath = folderPath
    this.data = data
    this.doFix = doFix
    this.doForce = doForce
    this.fileContent = ''
    this.originalFileContent = ''
    this.fileName = ''
    this.nbPassed = 0
    this.nbFailed = 0
  }

  async end () {
    await this.updateFile()
    await this.checkIssues()
  }

  async inspectFile (fileName) {
    this.fileName = fileName
    this.originalFileContent = this.fileContent = await readFileInFolder(this.folderPath, fileName, true)
  }

  async updateFile () {
    if (!this.doFix) return
    if (this.originalFileContent === this.fileContent) return
    return createFile(this.folderPath, this.fileName, this.fileContent)
  }

  async checkFileExists (fileName, justWarn = false) {
    let fileExists = await folderContainsFile(this.folderPath, fileName)
    if (!fileExists && this.doFix) {
      const fileContent = await this.createFile(fileName)
      fileExists = fileContent.length > 0
    }
    this.test(fileExists, `has a ${fileName} file`, justWarn)
    return !!fileExists
  }

  async checkNoFileExists (fileName, justWarn = false) {
    const fileExists = await folderContainsFile(this.folderPath, fileName)
    this.test(!fileExists, `has no ${fileName} file`, justWarn)
  }

  async createFile (fileName) {
    const template = await readFileInFolder(templatePath, fileName, true)
    const fileContent = fillTemplate(template, this.data)
    if (fileContent.length > 0) {
      await createFile(this.folderPath, fileName, fileContent)
      log.fix('created', fileName)
    } else {
      log.warn(`please provide a data file to be able to fix a "${fileName}" file`)
    }
    return fileContent
  }

  async checkIssues () {
    if (this.nbFailed > 0 && this.doFix) {
      if (this.doForce) return this.createFile(this.fileName)
      log.info('this file has at least one issue, if you want repo-checker to overwrite this file use --force')
    }
  }

  async getFileSizeInKo (filePath) {
    return getFileSizeInKo(path.join(this.folderPath, filePath))
  }

  /**
   * shouldContains verify that the target file contains specific strings
   * @param {string} name The name of the test
   * @param {regex} regex The regex that should validate the test
   * @param {number} nbMatchExpected The number of matches expected
   * @param {boolean} justWarn If this test is optional
   * @return a boolean which indicate if the content exists
   */
  shouldContains (name, regex, nbMatchExpected = 1, justWarn = false, helpMessage = '', canFix = false) {
    if (!regex) regex = new RegExp(name)
    const contentExists = this.checkContains(regex, nbMatchExpected)
    const fix = this.doFix && canFix && !contentExists
    name += (contentExists || fix) ? '' : ` -- ${helpMessage || regex.toString().replace(/\\/g, '')}`
    const message = `${this.fileName} ${!contentExists ? (justWarn ? 'could have' : 'does not have') : 'has'} ${name} `
    if (fix) {
      log.fix(message)
    } else {
      this.test(contentExists, message, justWarn)
    }
    return contentExists
  }

  couldContains (name, regex, nbMatchExpected = 1, helpMessage = '', canFix = false) {
    return this.shouldContains(name, regex, nbMatchExpected, true, helpMessage, canFix)
  }

  checkContains (regex, nbMatchExpected) {
    const matches = this.fileContent.match(regex)
    const nbMatch = (matches && matches.length) || 0
    if (nbMatchExpected && nbMatch !== nbMatchExpected) {
      log.debug(regex.toString().replace('\n', ''), 'matched', nbMatch, 'instead of', nbMatchExpected)
    }
    return (nbMatch === nbMatchExpected)
  }

  test (isValid, message, justWarn = false) {
    if (!isValid && !justWarn) {
      this.nbFailed++
    } else {
      this.nbPassed++
    }
    log.test(isValid, message, justWarn)
  }
}
