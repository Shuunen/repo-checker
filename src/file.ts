import { bgYellow, black, fillTemplate } from 'shuutils'
import { ProjectData, templatePath } from './constants'
import { log } from './logger'
import { fileExists, getFileSizeInKo, join, messageToCode, readFileInFolder, readableRegex, writeFile } from './utils'

const defaultAmount = 99

// eslint-disable-next-line no-restricted-syntax
export class FileBase {

  public passed: string[] = []

  public warnings: string[] = []

  public failed: string[] = []

  public folderPath = ''

  public data = new ProjectData()

  public canFix = false

  public canForce = false

  public fileContent = ''

  public fileName = ''

  private originalFileContent = ''

  // eslint-disable-next-line @typescript-eslint/max-params
  public constructor (folderPath = '', data: Readonly<ProjectData> = new ProjectData(), canFix = false, canForce = false) {
    this.folderPath = folderPath
    this.data = data
    this.canFix = canFix
    this.canForce = canForce
  }

  // eslint-disable-next-line complexity, sonarjs/cognitive-complexity, @typescript-eslint/max-params
  public shouldContains (name: string, regex?: Readonly<RegExp>, nbMatchExpected = defaultAmount, willJustWarn = false, helpMessage = '', canFix = false): boolean {
    // eslint-disable-next-line security/detect-non-literal-regexp
    const regexp = regex ?? new RegExp(name, 'u')
    const isOk = this.checkContains(regexp, nbMatchExpected)
    const willFix = this.canFix && canFix && !isOk
    let finalName = name
    finalName += isOk || willFix ? '' : ` -- ${helpMessage.length > 0 ? helpMessage : readableRegex(regexp)} ${canFix ? bgYellow(black('[ fixable ]')) : ''}`
    const have = willJustWarn ? 'could have' : 'does not have'
    const message = `${this.fileName} ${isOk ? 'has' : have} ${finalName} `
    if (willFix) log.fix(message)
    else this.test(isOk, message, willJustWarn)
    return isOk
  }

  // eslint-disable-next-line @typescript-eslint/max-params
  public couldContains (name: string, regex?: Readonly<RegExp>, nbMatchExpected = defaultAmount, helpMessage = '', canFix = false): boolean {
    return this.shouldContains(name, regex, nbMatchExpected, true, helpMessage, canFix)
  }

  public checkContains (regex: Readonly<RegExp>, nbMatchExpected = defaultAmount): boolean {
    // eslint-disable-next-line regexp/prefer-regexp-exec
    const matches = this.fileContent.match(regex) ?? []
    const hasExpectedMatches = nbMatchExpected === defaultAmount ? matches.length > 0 : nbMatchExpected === matches.length
    // eslint-disable-next-line @typescript-eslint/no-base-to-string
    if (!hasExpectedMatches) log.debug(regex.toString().replace('\n', ''), `matched ${matches.length} instead of ${nbMatchExpected === defaultAmount ? 'one or more' : nbMatchExpected}`)
    return hasExpectedMatches
  }

  public test (isValid: boolean, message: string, willJustWarn = false): boolean {
    const finalMessage = message.startsWith(this.fileName) ? message : `${this.fileName} ${message}`
    const code = messageToCode(finalMessage)
    if (isValid) { this.passed.push(code); log.test(isValid, finalMessage) }
    else if (willJustWarn) { this.warnings.push(code); log.warn(finalMessage) }
    else { this.failed.push(code); log.error(finalMessage) }
    return isValid
  }

  public couldContainsSchema (url: string): boolean {
    const line = `"$schema": "${url}",`
    const hasSchema = this.couldContains('a $schema declaration', /"\$schema": "/u, 1, `like ${line}`, true)
    if (hasSchema) return true
    if (!this.canFix) return hasSchema
    // eslint-disable-next-line prefer-named-capture-group, regexp/prefer-named-capture-group
    this.fileContent = this.fileContent.replace(/(^\{\n)(\s+)/u, `$1$2${line}\n$2`)
    return true
  }

  public async end (): Promise<void> {
    await this.updateFile()
    await this.checkIssues()
  }

  public async inspectFile (fileName: string): Promise<void> {
    this.fileName = fileName
    this.originalFileContent = await readFileInFolder(this.folderPath, fileName).catch(() => '')
    this.fileContent = this.originalFileContent
  }

  // eslint-disable-next-line max-statements
  public async updateFile (): Promise<void> {
    if (this.originalFileContent.trim() === this.fileContent.trim()) { log.debug('avoid file update when updated content is the same'); return }
    if (!this.canFix) { log.debug('cant update file if fix not active'); return }
    if (this.failed.length > 0 && !this.canForce) { log.debug('cant update file without force if some checks failed'); return }
    if (this.fileName.length === 0) { log.debug('cant update file without a file name, probably running tests'); return }
    await writeFile(join(this.folderPath, this.fileName), this.fileContent) // if you don't await, the file is updated after the end of the process and tests are failing
    log.debug('updated', this.fileName, 'with the new content')
  }

  public async fileExists (fileName: string): Promise<boolean> {
    return await fileExists(join(this.folderPath, fileName))
  }

  /**
   * Check if a file exists, create a file if fix enabled, trigger a success or fail
   * @param fileName the file name to check
   * @param willJustWarn just warn if the file is not found
   * @returns true if the file is found
   */
  public async checkFileExists (fileName: string, willJustWarn = false): Promise<boolean> {
    let hasFile = await this.fileExists(fileName)
    if (!hasFile && this.canFix) {
      const fileContent = await this.initFile(fileName)
      hasFile = fileContent.length > 0
    }
    this.test(hasFile, `has a ${fileName} file`, willJustWarn)
    return hasFile
  }

  public async checkNoFileExists (fileName: string): Promise<void> {
    const hasFile = await this.fileExists(fileName)
    this.test(!hasFile, `has no ${fileName} file`)
  }

  public async getFileSizeInKo (filePath: string): Promise<number> {
    return await getFileSizeInKo(join(this.folderPath, filePath))
  }

  public async initFile (fileName: string): Promise<string> {
    const template = await readFileInFolder(templatePath, fileName).catch(() => '')
    if (template === '') {
      log.debug(`found no template ${fileName}, using a empty string instead`)
      return ''
    }
    // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
    const fileContent = fillTemplate(template, this.data as unknown as Record<string, unknown>)
    if (fileContent.includes('{{')) log.warn(`please provide a data file to be able to fix a "${fileName}" file`)
    else void this.createFile(fileName, fileContent)
    return fileContent
  }

  // eslint-disable-next-line @typescript-eslint/require-await
  private async createFile (fileName: string, fileContent: string): Promise<void> {
    void writeFile(join(this.folderPath, fileName), fileContent)
    log.fix('created', fileName)
  }

  private async checkIssues (): Promise<void> {
    if (this.failed.length > 0 && this.canFix) {
      if (this.canForce) {
        await this.initFile(this.fileName)
        return
      }
      log.info('this file has at least one issue, if you want repo-checker to overwrite this file use --force')
    }
  }
}
