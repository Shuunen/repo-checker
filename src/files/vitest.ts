import { FileBase } from '../file.ts'

// eslint-disable-next-line no-export
export class VitestFile extends FileBase {
  // eslint-disable-next-line jsdoc/require-jsdoc
  // eslint-disable-next-line max-lines-per-function
  public async start() {
    if (!this.data.isUsingVitest) return
    const hasMtsFile = await this.fileExists('vitest.config.mts')
    const hasJsFile = await this.fileExists('vitest.config.js')
    const hasTsFile = await this.fileExists('vitest.config.ts')
    const hasMjsFile = await this.fileExists('vitest.config.mjs')
    if (hasMtsFile) await this.inspectFile('vitest.config.mts')
    /* c8 ignore next 3 */
    if (hasJsFile) await this.inspectFile('vitest.config.js')
    if (hasTsFile) await this.inspectFile('vitest.config.ts')
    if (hasMjsFile) await this.inspectFile('vitest.config.mjs')
    this.couldContains('a pool thread setting', /pool: 'threads'/u)
  }
}
