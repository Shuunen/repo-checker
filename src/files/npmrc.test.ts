import { expect, it } from 'vitest'
import { ProjectData, repoCheckerPath } from '../constants'
import { cleanInstanceForSnap, promiseTrue, promiseVoid, tsProjectFolder } from '../mock'
import { NpmRcFile } from './npmrc.file'

it('npm rc A file exists & hasTaskPrefix', async () => {
  const instance = new NpmRcFile(repoCheckerPath, new ProjectData({ hasTaskPrefix: true, isQuiet: true }))
  await instance.start()
  await instance.end()
  expect(cleanInstanceForSnap(instance)).toMatchSnapshot()
})

it('npm rc B file exists & !hasTaskPrefix', async () => {
  const instance = new NpmRcFile(repoCheckerPath, new ProjectData({ hasTaskPrefix: false, isQuiet: true }))
  await instance.start()
  await instance.end()
  expect(cleanInstanceForSnap(instance)).toMatchSnapshot()
})

it('npm rc C file !exists & hasTaskPrefix', async () => {
  const instance = new NpmRcFile(tsProjectFolder, new ProjectData({ hasTaskPrefix: true, isQuiet: true }))
  await instance.start()
  await instance.end()
  expect(cleanInstanceForSnap(instance)).toMatchSnapshot()
})

it('npm rc D fix file', async () => {
  const instance = new NpmRcFile('', new ProjectData({ hasTaskPrefix: true, isQuiet: true }), true)
  instance.fileExists = promiseTrue
  instance.inspectFile = promiseVoid
  instance.fileContent = ''
  await instance.start()
  await instance.end()
  expect(cleanInstanceForSnap(instance)).toMatchSnapshot()
})
