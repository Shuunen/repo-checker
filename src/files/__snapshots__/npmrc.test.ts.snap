// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`npm rc A file exists & hasTaskPrefix 1`] = `
{
  "canFix": false,
  "canForce": false,
  "data": {
    "canAutoMergeDeps": true,
    "hasMainBranch": false,
    "hasTaskPrefix": true,
    "isCli": false,
    "isModule": false,
    "isPublishedPackage": false,
    "isQuiet": true,
    "isUsingC8": false,
    "isUsingDependencyCruiser": false,
    "isUsingEslint": false,
    "isUsingNyc": false,
    "isUsingOxc": false,
    "isUsingShuutils": false,
    "isUsingTailwind": false,
    "isUsingTypescript": false,
    "isUsingV8": false,
    "isUsingVue": false,
    "isWebPublished": false,
    "license": "GPL-3.0",
    "maxSizeKo": 1,
    "packageName": "",
    "repoId": "a-great-repo",
    "shouldAvoidSass": true,
    "userId": "Shuunen",
    "userIdLowercase": "shuunen",
    "userMail": "romain.racamier@gmail.com",
    "userName": "Romain Racamier-Lafon",
    "webUrl": "https://my-website.com",
    "willGenerateReport": true,
  },
  "failed": [],
  "fileContent": "enable-pre-post-scripts=true
",
  "fileName": ".npmrc",
  "originalFileContent": "enable-pre-post-scripts=true
",
  "passed": [
    "has-a-npmrc-file",
    "npmrc-has-enable-pre-post-scripts-option",
  ],
  "warnings": [],
}
`;

exports[`npm rc B file exists & !hasTaskPrefix 1`] = `
{
  "canFix": false,
  "canForce": false,
  "data": {
    "canAutoMergeDeps": true,
    "hasMainBranch": false,
    "hasTaskPrefix": false,
    "isCli": false,
    "isModule": false,
    "isPublishedPackage": false,
    "isQuiet": true,
    "isUsingC8": false,
    "isUsingDependencyCruiser": false,
    "isUsingEslint": false,
    "isUsingNyc": false,
    "isUsingOxc": false,
    "isUsingShuutils": false,
    "isUsingTailwind": false,
    "isUsingTypescript": false,
    "isUsingV8": false,
    "isUsingVue": false,
    "isWebPublished": false,
    "license": "GPL-3.0",
    "maxSizeKo": 1,
    "packageName": "",
    "repoId": "a-great-repo",
    "shouldAvoidSass": true,
    "userId": "Shuunen",
    "userIdLowercase": "shuunen",
    "userMail": "romain.racamier@gmail.com",
    "userName": "Romain Racamier-Lafon",
    "webUrl": "https://my-website.com",
    "willGenerateReport": true,
  },
  "failed": [],
  "fileContent": "enable-pre-post-scripts=true
",
  "fileName": ".npmrc",
  "originalFileContent": "enable-pre-post-scripts=true
",
  "passed": [
    "has-a-npmrc-file",
  ],
  "warnings": [],
}
`;

exports[`npm rc C file !exists & hasTaskPrefix 1`] = `
{
  "canFix": false,
  "canForce": false,
  "data": {
    "canAutoMergeDeps": true,
    "hasMainBranch": false,
    "hasTaskPrefix": true,
    "isCli": false,
    "isModule": false,
    "isPublishedPackage": false,
    "isQuiet": true,
    "isUsingC8": false,
    "isUsingDependencyCruiser": false,
    "isUsingEslint": false,
    "isUsingNyc": false,
    "isUsingOxc": false,
    "isUsingShuutils": false,
    "isUsingTailwind": false,
    "isUsingTypescript": false,
    "isUsingV8": false,
    "isUsingVue": false,
    "isWebPublished": false,
    "license": "GPL-3.0",
    "maxSizeKo": 1,
    "packageName": "",
    "repoId": "a-great-repo",
    "shouldAvoidSass": true,
    "userId": "Shuunen",
    "userIdLowercase": "shuunen",
    "userMail": "romain.racamier@gmail.com",
    "userName": "Romain Racamier-Lafon",
    "webUrl": "https://my-website.com",
    "willGenerateReport": true,
  },
  "failed": [],
  "fileContent": "",
  "fileName": "",
  "originalFileContent": "",
  "passed": [],
  "warnings": [
    "has-a-npmrc-file",
  ],
}
`;

exports[`npm rc D fix file 1`] = `
{
  "canFix": true,
  "canForce": false,
  "data": {
    "canAutoMergeDeps": true,
    "hasMainBranch": false,
    "hasTaskPrefix": true,
    "isCli": false,
    "isModule": false,
    "isPublishedPackage": false,
    "isQuiet": true,
    "isUsingC8": false,
    "isUsingDependencyCruiser": false,
    "isUsingEslint": false,
    "isUsingNyc": false,
    "isUsingOxc": false,
    "isUsingShuutils": false,
    "isUsingTailwind": false,
    "isUsingTypescript": false,
    "isUsingV8": false,
    "isUsingVue": false,
    "isWebPublished": false,
    "license": "GPL-3.0",
    "maxSizeKo": 1,
    "packageName": "",
    "repoId": "a-great-repo",
    "shouldAvoidSass": true,
    "userId": "Shuunen",
    "userIdLowercase": "shuunen",
    "userMail": "romain.racamier@gmail.com",
    "userName": "Romain Racamier-Lafon",
    "webUrl": "https://my-website.com",
    "willGenerateReport": true,
  },
  "failed": [],
  "fileContent": "
enable-pre-post-scripts=true",
  "fileName": "",
  "originalFileContent": "",
  "passed": [
    "has-a-npmrc-file",
  ],
  "warnings": [
    "could-have-enable-pre-post-scripts-option-enable-pre-post-scriptstrue",
  ],
}
`;
